src/gluonts/dataset/split/splitter.py

```diff
@@ -129,14 +152,21 @@ def __getitem__(self, slice_: slice) -> "TimeSeriesSlice":
         feat_dynamic_cat = None
 
         if self.feat_dynamic_real is not None:
-            feat_dynamic_real = [feat for feat in self.feat_dynamic_real]
+            feat_dynamic_real = [
+                feat[slice_] for feat in self.feat_dynamic_real
+            ]
 
         if self.feat_dynamic_cat is not None:
-            feat_dynamic_cat = [feat for feat in self.feat_dynamic_cat]
+            feat_dynamic_cat = [feat[slice_] for feat in self.feat_dynamic_cat]
+
+        target = self.target[slice_]
+
+        assert all([len(target) == len(i) for i in feat_dynamic_real])
```

Comment:
Don't use `i` if it is not an integer.

```suggestion
        assert all([len(target) == len(feat) for feat in feat_dynamic_real])
```